engine/examples/client/pike/main.go

```diff
@@ -0,0 +1,50 @@
+package main
+
+import (
+	"encoding/hex"
+	"fmt"
+
+	"github.com/libsv/go-bk/bec"
+
+	"github.com/bitcoin-sv/spv-wallet/engine/pike"
+	"github.com/bitcoin-sv/spv-wallet/engine/script/template"
+)
+
+func main() {
+	// Example sender's public key (replace with actual sender's public key)
+	// generating keys
+	senderPublicKeyHex := "034252e5359a1de3b8ec08e6c29b80594e88fb47e6ae9ce65ee5a94f0d371d2cde"
+	senderPublicKeyBytes, err := hex.DecodeString(senderPublicKeyHex)
+	if err != nil {
+		panic(err)
+	}
+	senderPubKey, err := bec.ParsePubKey(senderPublicKeyBytes, bec.S256())
+	if err != nil {
+		panic(err)
+	}
+
+	receiverPublicKeyHex := "027c1404c3ecb034053e6dd90bc68f7933284559c7d0763367584195a8796d9b0e"
+	receiverPublicKeyBytes, err := hex.DecodeString(receiverPublicKeyHex)
+	if err != nil {
+		panic(err)
+	}
+	receiverPubKey, err := bec.ParsePubKey(receiverPublicKeyBytes, bec.S256())
+	if err != nil {
+		panic(err)
+	}
+
+	// example of usage pike2
+	outputsTemplate, _ := template.P2PKH(10000)
+	fmt.Println(outputsTemplate)
+
+	lockingScripts, err := pike.GenerateLockingScriptsFromTemplates(outputsTemplate, senderPubKey, receiverPubKey, "reference")
```

Comment:
It would be cool to have similar example (as testable example) for `template` package
so first create P2PKH template, and then call template.Evaluate on it with some public key.

Reply:
but we need 2 pub keys for this, anyway have look as i make this testable example